src/valid/expression.rs

```diff
@@ -992,6 +991,22 @@ impl super::Validator {
                             _ => return Err(ExpressionError::InvalidArgumentType(fun, 0, arg)),
                         }
                     }
+                    Mf::Sign => {
+                        if arg1_ty.is_some() | arg2_ty.is_some() | arg3_ty.is_some() {
```

Comment:
question(blocking): Is there a reason we're using the bit-or operation (`|`) instead of the short-circuiting version (`||`)?

```suggestion
                        if arg1_ty.is_some() || arg2_ty.is_some() || arg3_ty.is_some() {
```

Reply:
Thanks! There was no reason, created https://github.com/gfx-rs/naga/pull/2465 to fix this in other existing places in the same file.